transfer_nlp/plugins/config.py

```diff
@@ -246,17 +247,18 @@ def resolve_simple_value(factory_key: str, val: Any) -> Any:
         if not clazz:
             raise UnknownPluginException(object_dict["_name"])
 
-        spec = inspect.getfullargspec(clazz.__init__)
+        spec = inspect.getfullargspec(clazz)
         params = {}
         param2config_key = {}
         named_params = {p: pv for p, pv in object_dict.items() if p != '_name'}
         default_params = {p: pv for p, pv in zip(reversed(spec.args), reversed(spec.defaults))} if spec.defaults else {}
+        spec_args = spec.args[1:] if 'self' in spec.args else spec.args
```

Comment:
i don't think this check is quite right. there's nothing special about the `self` param name. the special thing is the first argument of instance methods refers to the instance itself. you can name it anything you want, though people almost always call it self and then don't use self for functions. 

but we should improve the check anyways. :)

Reply:
Added a check for classes, methods or functions